tests/unittest.c

```diff
@@ -27,19 +27,26 @@ int dumper(void* user, const char* section, const char* name,
 #endif
 {
     User = *((int*)user);
-    if (strcmp(section, Prev_section)) {
+    if (!name || strcmp(section, Prev_section)) {
         printf("... [%s]\n", section);
         strncpy(Prev_section, section, sizeof(Prev_section));
         Prev_section[sizeof(Prev_section) - 1] = '\0';
     }
 
 #if INI_HANDLER_LINENO
-    printf("... %s=%s;  line %d\n", name, value, lineno);
+	if(name) {
+		printf("... %s=%s;  line %d\n", name, value, lineno);
+	} else {
+		/* No name means that this was called for a section header */
```

Comment:
I don't think this section is used (and will go away if we do the above anyway).